### PR TITLE
[1pt] PR: Tool to collate catchment attributes

### PR DESCRIPTION
**Pull request by CalebOliven-NOAA**
`collate_catchment_attributes.py` is located in tools. It is a simple script to assemble desired attributes for further study. The code loops through huc directories and pulls out static (does not change with stage) attributes by hydroid. It then links to sierra test metrics via location id and hydro id. 

## Additions

-new tool to collate catchment attributes
-`/tools/collate_catchment_attributes.py`

## Testing

these are the files I used to test the code, will run much faster
sierra_test = `"/data/outputs/carson_test1/sierra_test_sinuosity/usgs_gages_stats.gpkg"`
elev_link_table =`"/data/outputs/carson_test1/sierra_test/agg_usgs_elev_table.csv"`
hydrotable_directory =`"/data/outputs/carson_test1"`

`python collate_catchment_attributes.py -d "/data/outputs/carson_test1" -s "/data/outputs/carson_test1/sierra_test_sinuosity/usgs_gages_stats.gpkg" -o "/data/temp/caleb/test_collate_date.csv" -l "/data/outputs/carson_test1/sierra_test/agg_usgs_elev_table.csv"`

## Notes

code to run the file with current inputs `python collate_catchment_attributes.py -d "/data/previous_fim/fim_3_0_24_14_ms/" -s "/data/tools/sierra_test/official_versions/fim_3_0_24_14_ms/usgs_gages_stats.gpkg" -o "/data/temp/caleb/218collate.csv" -l "/data/tools/sierra_test/official_versions/fim_3_0_24_14_ms/agg_usgs_elev_table.csv"`

the script is called `collate_catchment_attributes.py`
the inputs for the command are `collate_catchment_attributes(fim_directory, sierra_test_input, output_csv_destination, link_elev_table)`

the recommended fim directory at this time is `"/data/tools/sierra_test/official_versions/fim_3_0_24_14_ms/usgs_gages_stats.gpkg"`

the recommended sierra_test_input at this time is `/data/tools/sierra_test/official_versions/fim_3_0_24_14_ms/usgs_gages_stats.gpkg"`

the recommended link_elev_table at this time is `"/data/previous_fim/fim_3_0_24_14_ms/"`

## Todos

-

### CODE REVIEW

**Review by CarsonPruitt-NOAA — COMMENTED** [review]
Made a couple of code aesthetics comments on `collate_catchement_attributes.py`

**Comment by BradfordBates-NOAA** [comment]
Nice work, @CalebOliven-NOAA! A couple of remaining items before merging:

- [ ] Address Carson's comments above (remove extra spaces between imports)
- [ ] Add docstrings to code
- [ ] Change function names to snake_case. For example, ` importLinkTable` will become `import_link_table`
- [ ] Add 2 returns before function definitions

File: tools/collate_catchment_attributes.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    parser.add_argument('-l','--link-elev-table',help='elev table for linking sierra tests to hydrotable on locationid and hydroid',required=True)
+    
+    args = vars(parser.parse_args())
+
+    fim_directory = args['fim_directory']
+    sierra_test_input = args['sierra_test_input']
+    output_csv_destination = args['output_csv_destination']
+    link_elev_table = args['link_elev_table']
+    
+    aggregate_df = performMerge(assembleSierraTest(sierra_test_input),importLinkTable(link_elev_table),aggregateHydroTables(fim_directory))
```
[the remark this answers is not in the document]
Comment (CalebOliven-NOAA) [reply tools/collate_catchment_attributes.py]: cleaned

**Comment by CalebOliven-NOAA** [comment]
made requested changes.

**Review by BradfordBates-NOAA — APPROVED** [review]

**Pull request merged**